### PR TITLE
Implement ProtoCube and ImposterProtoCube

### PR DESCRIPTION
**Pull request by CursedFlames**

### CODE REVIEW

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java [hunk 1]
```diff
@@ -1,10 +1,191 @@
...
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "setCarvingMask(Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;Lnet/minecraft/world/level/chunk/CarvingMask;)V")
+    @Override public native void setCarvingMask(GenerationStep.Carving pStep, CarvingMask pCarvingMask);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "setLightEngine(Lnet/minecraft/world/level/lighting/LevelLightEngine;)V")
+    @Override public native void setLightEngine(LevelLightEngine pLightEngine);
+
+    @Override public void setBelowZeroRetrogen(@Nullable BelowZeroRetrogen pBelowZeroRetrogen) {
+        // Unused
+    }
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java]: Thoughts on making this dasm?

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java [hunk 2]
```diff
@@ -1,10 +1,191 @@
...
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "unpackTicks(Lnet/minecraft/world/ticks/ProtoChunkTicks;)Lnet/minecraft/world/ticks/LevelChunkTicks;")
+    private static <T> LevelChunkTicks<T> unpackTicks(ProtoChunkTicks<T> pTicks) {
+        return new LevelChunkTicks(pTicks.scheduledTicks());
+    }
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java]: Is there a reason this is dasm and defined, and not just native?

**Review by NotStirred — CHANGES_REQUESTED** [review]
Minor nits, very nice

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java (hunk 1 above)
Comment (CursedFlames) [reply src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java]: The original method body actually sets it; I'm replacing it with an empty method body. Will add a comment to make that clearer

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java (hunk 2 above)
Comment (CursedFlames) [reply src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java]: Good catch, missed that method body when I was adding dasm'd static methods. Will remove

**Review by NotStirred — APPROVED** [review]

**Pull request merged**